actions/treasures.go

```diff
@@ -31,26 +26,39 @@ type treasureRes struct {
 	Success string `json:"success"`
 }
 
-// Constant API connection to IOTA
-var IOTA_API *giota.API
-
-func init() {
-	host_ip := os.Getenv("HOST_IP")
-	if len(host_ip) >= 0 {
-		host_ip = strings.Trim(host_ip, "http://")
-		host_ip = strings.Trim(host_ip, "https://")
-	} else {
-		host_ip = "localhost"
-	}
-	provider := fmt.Sprintf("http://%s:14265/", host_ip)
-	IOTA_API = giota.NewAPI(provider, nil)
-}
-
 // Verifies the treasure and claims such treasure.
 func (t *TreasuresResource) VerifyAndClaim(c buffalo.Context) error {
 	req := treasureReq{}
 	oyster_utils.ParseReqBody(c.Request(), &req)
 
+	addr := models.ComputeSectorDataMapAddress(req.GenesisHash, req.SectorIdx, req.NumChunks)
+	iotaAddr := make([]giota.Address, len(addr))
+
+	for i, address := range addr {
+		iotaAddr[i] = giota.Address(address)
+	}
+
+	transactionsMap := services.FindTransactions(iotaAddr)
+
+	if len(transactionsMap) != len(iotaAddr) {
+		// indicate that PoW failure.
+	}
+
+	passedTimestamp := time.Now().AddDate(-1, 0, 0)
+	for _, iotaAddress := range iotaAddr {
+		if _, hasKey := transactionsMap[iotaAddress]; !hasKey {
+			// indicate that PoW failure
+		}
+
+		transactions := transactionsMap[iotaAddress]
+		// Check all the transactions has submit within the passed 1 year.
+		for _, transaction := range transactions {
```

Comment:
Could maybe do something like this:  have a bool powFailed := false before, and then set it to true if it finds at least one transaction per address that satisfies the timestamp check.  Otherwise, by the time the loop is done, powFailed would still be false, and so we know the PoW failed.

Reply:
Got it.